src/units/unit.cpp

```diff
@@ -1096,8 +1096,13 @@ color_t unit::xp_color() const
 	const bool far_advance  = static_cast<int>(experience_to_advance()) <= game_config::kill_experience*3;
 
 	color_t color = normal_color;
-
-	if(advances_to().size()){
+	bool hidden_amla = false;
+	for (const config& adv : get_modification_advances()){
+if(adv[ "hidden_amla" ].to_bool ()){
+hidden_amla = true;
+}
+	}
+if(advances_to().size() || hidden_amla == true){
```

Comment:
You need to fix your indentation here.

Also, remove the extra spaces after `for` and inside the square brackets.

Finally, I don't think this'll quite work how you want. It seems like specifying the key for one AMLA will actually apply it to that AMLA plus any AMLAs that appear after it in the unit type config. I'd suggest eliminating the boolean variable entirely.